pkg/frontend/nvme.go

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Intel Corporation
+
+// Package frontend implements the FrontEnd APIs (host facing) of the storage Server
+package frontend
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+
+	pb "github.com/opiproject/opi-api/storage/v1alpha1/gen/go"
+	"github.com/opiproject/opi-spdk-bridge/pkg/models"
+	"github.com/ulule/deepcopier"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+const (
+	mevTransportType = "npi"
+)
+
+type mevNvmfSubsystemAddListenerParams struct {
+	Nqn           string `json:"nqn"`
+	ListenAddress struct {
+		Trtype string `json:"trtype"`
+		Traddr string `json:"traddr"`
+	} `json:"listen_address"`
+}
+
+// CreateNVMeController creates an NVMe Controller
+func (s *Server) CreateNVMeController(ctx context.Context, in *pb.CreateNVMeControllerRequest) (*pb.NVMeController, error) {
```

Comment:
what is different about this function VS op-spdk-bridge impl ?

Reply:
opi-spdk-bridge works with `tcp` transport only at the moment. For this bridge we use another transport, and another format for the arguments.
The interface of the call is the same as in SPDK

Reply:
I would try to override `Trtype` and `Traddr` , then we can re-use all the code...

Reply:
Just wanted to note that by design, the RPC interfaces are exactly the same as regular SPDK NVMe-oF target and we treat it as a new transport type. So it really is just overriding trtype/traddr.

Reply:
Yep, what I'm saying, we can maybe place trtype/traddr in the server struct and then replacement will be easier without re-writing the functions, just init the server with proper arguments...